cherrypick: sql: allow listing orphan dropped tables in crdb_internal.tables

knz opened the pull request:
The virtual table `crdb_internal.tables` purposefully lists all table
descriptors, including dropped descriptors, for debugging purposes.

Prior to this patch, an error was emitted when a dropped descriptor
didn't have a valid parent database. The situation should not cause an
error, however, as a dropped table descriptor can exist with no
parent, e.g. as a consequence of DROP DATABASE CASCADE.

This patch fixes the issue by listing the parent ID of a dropped table
in numeric form if the database descriptor cannot be found.

Cherry-pick of #18572.

cc @cockroachdb/release

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18574)
<!-- Reviewable:end -->

knz commented:
rebased after the fix to #18572

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18574#-:-KuQU90u30JQGqbwn0tA:bnfp4nl)*
<!-- Sent from Reviewable.io -->

knz committed:
sql: allow listing orphan dropped tables in crdb_internal.tables

The virtual table `crdb_internal.tables` purposefully lists all table
descriptors, including dropped descriptors, for debugging purposes.

Prior to this patch, an error was emitted when a dropped descriptor
didn't have a valid parent database. The situation should not cause an
error, however, as a dropped table descriptor can exist with no
parent, e.g. as a consequence of DROP DATABASE CASCADE.

This patch fixes the issue by listing the parent ID of a dropped table
in numeric form if the database descriptor cannot be found.
